### ISSUE TITLE
[BUG] Simple einsum program failed to compile

### ISSUE THREAD

**Issue by lucifer1004** [NVIDIA/MatX#597]
The following code could not compile

```cpp
#include "matx.h"

int main([[maybe_unused]] int argc, [[maybe_unused]] char **argv) {
  MATX_ENTER_HANDLER();

  auto a = matx::make_tensor<float>({10, 10});
  auto b = matx::make_tensor<float>({10, 10});
  auto c = matx::make_tensor<float>({10, 10});

  auto exec = matx::cudaExecutor();
  (c = matx::cutensor::einsum("ik,kl->il", a, b)).run(exec);
  matx::print(c);

  CUDA_CHECK_LAST_ERROR();
  MATX_EXIT_HANDLER();
}
```

The error message was:

```cpp
MatX/include/matx/transforms/einsum.h(457): error: class "matx::detail::matxCache_t" may not have a template argument list
  static matxCache_t<EinsumParams_t<InT...>, EinsumParamsKeyHash<InT...>, EinsumParamsKeyEq<InT...>> einsum_cache;
```

**Comment by lucifer1004** [comment]
It seems that the code for einsum cache has not been updated according to `matxCache_t`'s API changes.

**Comment by tmartin-gh** [comment]
Thanks!  Reproduced when compiling with MATX_EN_CUTENSOR=ON.  Secondary bug: compilation succeeds when using MATX_EN_CUTENSOR=OFF, but we should probably fail to compile if MATX_EN_CUTENSOR=OFF and encounter usage of matx::cutensor namespace...

**Comment by tmartin-gh** [comment]
Note: Example reproducer "ik,kl->il" is a matrix multiply, should have been caught by TYPED_TEST(EinsumTestsFloatNonComplexNonHalfTypes, GEMM).

**Comment by cliffburdick** [comment]
Fixed by [2c34e34](https://github.com/NVIDIA/MatX/commit/2c34e34857d4e2a9f134b9170472b6398e79c93e)
